sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -101,17 +101,17 @@ internal TransportProducerPool()
                 return new PooledProducer(EventHubProducer);
             }
 
-            var identifier = Guid.NewGuid().ToString();
-            var item = Pool.GetOrAdd(partitionId, id => new PoolItem(partitionId, TransportProducerFactory(id), removeAfterDuration));
+            var identifier = Guid.NewGuid();
+            var item = Pool.GetOrAdd(partitionId, id => new PoolItem(id, TransportProducerFactory(id), removeAfterDuration));
```

Comment:
Doesn't changing the first arg from `partitionId` to `id` here also change the semantics?   That makes me a bit nervous.   Then again, looking at the constructor for `PoolItem` it would seem they need to be the same value so that the factory and item agree on what partition they're for...  why in the world do we have two of them?

Reply:
The partition key is used as the key to the dictionary which is also what is being passed as an argument into the value factory method. In an ideal world the value factory method would not need to reference anything outside the closure so I did a first step to align things here since partionKey and id are the same

Reply:
ahh...  yeah.   Reading is hard.